src/components/VersionFileViewer/index.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+
+import AccordionMenu, { AccordionItem } from '../AccordionMenu';
+import CodeOverview from '../CodeOverview';
+import FileMetadata from '../FileMetadata';
+import FileTree, { PublicProps as FileTreeProps } from '../FileTree';
+import { ContentShell } from '../FullscreenGrid';
+import KeyboardShortcuts from '../KeyboardShortcuts';
+import Loading from '../Loading';
+import { Version, VersionFile } from '../../reducers/versions';
+import { AnyReactNode } from '../../typeUtils';
+import { gettext } from '../../utils';
+import styles from './styles.module.scss';
+
+export enum ItemTitles {
+  Files = 'Files',
+  Information = 'Information',
+  // eslint-disable-next-line no-shadow
+  KeyboardShortcuts = 'Keyboard Shortcuts',
+}
+
+export type PublicProps = {
+  children: AnyReactNode;
+  file: VersionFile | null | void;
+  onSelectFile: FileTreeProps['onSelect'];
+  showFileInfo: boolean;
+  version: Version | void | null;
+};
+
+const VersionFileViewer = ({
+  children,
+  file,
+  onSelectFile,
+  showFileInfo,
+  version,
+}: PublicProps) => {
+  if (!version) {
+    return (
+      <ContentShell>
+        <Loading message={gettext('Loading version...')} />
+      </ContentShell>
+    );
+  }
+
+  return (
+    <ContentShell
+      mainSidePanel={
+        <AccordionMenu>
+          <AccordionItem expandedByDefault title={ItemTitles.Files}>
+            <FileTree onSelect={onSelectFile} versionId={version.id} />
+          </AccordionItem>
+          {showFileInfo ? (
+            <AccordionItem title={ItemTitles.Information}>
+              {file ? (
+                <FileMetadata file={file} />
+              ) : (
+                <Loading message={gettext('Loading file...')} />
+              )}
+            </AccordionItem>
+          ) : null}
+          <AccordionItem title={ItemTitles.KeyboardShortcuts}>
+            <KeyboardShortcuts
+              currentPath={version.selectedPath}
+              versionId={version.id}
+            />
+          </AccordionItem>
+        </AccordionMenu>
+      }
+      mainSidePanelClass={styles.mainSidePanel}
+      altSidePanel={
+        file ? <CodeOverview content={file.content} version={version} /> : null
```

Comment:
FYI, I think the "view an image" patch that I'm about to land will cause a conflict with this.